### ISSUE TITLE
dependency: don't use dvcignore

### ISSUE THREAD

**Issue by efiop** [iterative/dvc#5256]
```
def test_run_dvcignored_dep(tmp_dir, dvc, run_copy):             
    tmp_dir.gen(                                                 
        {                                                        
            ".dvcignore": "dir\n",                               
            "dir": {"foo": "foo"},                               
        }                                                        
    )                                                            
    run_copy(                                                    
        os.path.join("dir", "foo"), "bar", name="copy-foo-to-bar"
    )                                                            
    assert (tmp_dir / "bar").read_text() == "foo"                
```

results in

```
/home/efiop/git/dvc/tests/dir_helpers.py:332: in run_copy                                                 
    return dvc.run(                                                                                       
/home/efiop/git/dvc/dvc/repo/__init__.py:49: in wrapper                                                   
    return f(repo, *args, **kwargs)                                                                       
/home/efiop/git/dvc/dvc/repo/scm_context.py:4: in run                                                     
    result = method(repo, *args, **kw)                                                                    
/home/efiop/git/dvc/dvc/repo/run.py:113: in run                                                           
    stage = create_stage(                                                                                 
/home/efiop/git/dvc/dvc/stage/__init__.py:84: in create_stage                                             
    fill_stage_dependencies(                                                                              
/home/efiop/git/dvc/dvc/stage/utils.py:97: in fill_stage_dependencies                                     
    stage.deps += loads_from(stage, deps or [], erepo=erepo)                                              
/home/efiop/git/dvc/dvc/dependency/__init__.py:95: in loads_from                                          
    return [_get(stage, s, info.copy()) for s in s_list]                                                  
/home/efiop/git/dvc/dvc/dependency/__init__.py:95: in <listcomp>                                          
    return [_get(stage, s, info.copy()) for s in s_list]                                                  
/home/efiop/git/dvc/dvc/dependency/__init__.py:81: in _get                                                
    return LocalDependency(stage, p, info)                                                                
/home/efiop/git/dvc/dvc/output/local.py:24: in __init__                                                   
    super().__init__(stage, path, *args, **kwargs)                                                        
/home/efiop/git/dvc/dvc/output/base.py:117: in __init__                                                   
    self._validate_output_path(path, stage)                                                               
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                                                                                                          
cls = <class 'dvc.dependency.local.LocalDependency'>, path = 'dir/foo', stage = Stage: 'copy-foo-to-bar'  
                                                                                                          
    @classmethod                                                                                          
    def _validate_output_path(cls, path, stage=None):                                                     
        from dvc.dvcfile import is_valid_filename                                                         
                                                                                                          
        if is_valid_filename(path):                                                                       
            raise cls.IsStageFileError(path)                                                              
                                                                                                          
        if stage:                                                                                         
            check = stage.repo.tree.dvcignore.check_ignore(path)                                          
            if check.match:                                                                               
>               raise cls.IsIgnoredError(check)                                                           
E               dvc.output.base.OutputIsIgnoredError: Path 'dir/foo' is ignored by                        
E               .dvcignore:1:dir                                                                          
                                                                                                          
/home/efiop/git/dvc/dvc/output/base.py:566: OutputIsIgnoredError                                          
```

because we use dvcignore in dependency tree, while we probably shouldn't.

https://discord.com/channels/485586884165107732/485596304961962003/798656464646307852

CC @PeterFogh

**Comment by skshetry** [comment]
Also, we need #4986 to fix this (thanks to @karajan1001), as we shouldn't be doing `check_ignore` there.

**Comment by karajan1001** [comment]
@efiop @skshetry 
I will solve myself #4986 today.

**Comment by efiop** [comment]
For the record: the original issue is not fixed.

**Comment by karajan1001** [comment]
> For the record: the original issue is not fixed. Having dvcignore built into LocalFileSystem is no longer suitable, esp with fsspec migration. I would consider using it explicitly to check if something is ignored, kinda like `git add` does, instead of having an implicit behaviour.

I'll look into it tomorrow.

**Comment by karajan1001** [comment]
Seems that it is because our `LocalDependency ` inherit from `LocalOutput `.

https://github.com/iterative/dvc/blob/e0a0342ab3594c1594e12086619fd81ebb5550ef/dvc/dependency/local.py#L5

Create a local path that didn't use `dvcignore`, and both  `LocalDependency ` and `LocalOutput ` inherit from it?

**Comment by efiop** [comment]
@karajan1001 That could also be an option, indeed. We've been meaning to stop inheriting from Output for a long time, but that might be a bit much to do right now. We might get away with checking self.IS_DEPENDENCY when creating fs instance and when checking `_validate_output_path`. Need to give it a try.

**Comment by karajan1001** [comment]
already fixed by #5812 , related tests.
https://github.com/iterative/dvc/blob/b349b39d6266a28705b03b1b06e5c0cf98a9080d/tests/func/test_ignore.py#L417-L420
